ctl/string.h

```diff
@@ -220,8 +285,69 @@ struct string
     {
         return compare(s) >= 0;
     }
+
+  private:
+    inline bool isbig() const noexcept
+    {
+        return *(__builtin_launder(blob) + __::sso_max) & 0x80;
```

Comment:
move and forward are both trivial plain C++ functions; there is no reason we couldn't write a brief `ctl::move` and `ctl::forward` of our own.

E.g. capnproto's libkj does this:

https://github.com/capnproto/capnproto/blob/5faf3144b5e1b0fbbe80992f7e3a890a472c82a3/c%2B%2B/src/kj/common.h#L678-L687

`std::launder` is different. It is an escape hatch from the compiler's ability to make crazy assumptions based on the object lifetime rules of C++ - I have not carefully studied this in a bit but my understanding is, for example, if you have a byte that is part of a union where one arm is `const` and the other isn't, and you change it in the non-`const` arm, then unless you call `launder` when referencing the `const` arm, the compiler can just assume that the value did not change and not bother to look.

Since you obviously can't implement this in terms of the actual C++ language itself, there is going to be a `__builtin_launder` or something that your STL provider is going to collaborate with your compiler author to use.

Reply:
Thank you. This information has much more clarity than Google results.